src/Confluent.SchemaRegistry.Serdes.Avro/SpecificSerializerImpl.cs

```diff
@@ -115,39 +169,61 @@ internal class SpecificSerializerImpl<T> : IAvroSerializerImpl<T>
                 );
             }
 
-            avroWriter = new SpecificWriter<T>(writerSchema);
-            writerSchemaString = writerSchema.ToString();
+            specificSerializerSchemaData.AvroWriter = new SpecificWriter<T>(specificSerializerSchemaData.WriterSchema);
+            specificSerializerSchemaData.WriterSchemaString = specificSerializerSchemaData.WriterSchema.ToString();
+            return specificSerializerSchemaData;
         }
 
         public async Task<byte[]> Serialize(string topic, T data, bool isKey)
         {
             try
-            {
+            {   
+                SpecificSerializerSchemaData currentSchemaData;
                 await serializeMutex.WaitAsync().ConfigureAwait(continueOnCapturedContext: false);
                 try
                 {
+                    if (_isMultipleSchemas)
+                    {
+                        var key = data.GetType();
+                        if (!schemaData.TryGetValue(key, out currentSchemaData))
+                        {
+                            currentSchemaData = ExtractSchemaData(key);
+                            schemaData[key] = currentSchemaData;
+                        }
+                    }
+                    else
+                    {
+                        currentSchemaData = schemaData.Values.First();
+                    }
+                    
+                    
                     string fullname = null;
-                    if (data is ISpecificRecord && ((ISpecificRecord)data).Schema is Avro.RecordSchema)
+                    if (data is ISpecificRecord && ((ISpecificRecord) data).Schema is Avro.RecordSchema)
                     {
-                        fullname = ((Avro.RecordSchema)((ISpecificRecord)data).Schema).Fullname;
+                        fullname = ((Avro.RecordSchema) ((ISpecificRecord) data).Schema).Fullname;
                     }
 
                     string subject = this.subjectNameStrategy != null
                         // use the subject name strategy specified in the serializer config if available.
-                        ? this.subjectNameStrategy(new SerializationContext(isKey ? MessageComponentType.Key : MessageComponentType.Value, topic), fullname)
+                        ? this.subjectNameStrategy(
```

Comment:
more whitespace only change - fix them all, even ones i might have missed, because I won't in a later pass.